src/wxflow/scheduler/scheduler.py

```diff
@@ -0,0 +1,189 @@
+
+        """
+
+        # Cache incoming config
+        self._config = AttrDict(config)
+
+        self.specs = self._config_to_specs
+        self.batch_card = []
+
+    @property
+    def _config_to_specs(self) -> AttrDict:
+        """
+        Converts the internal configuration dictionary to a standardized specification format.
+
+        This method processes the internal `_config` attribute and returns a deep copy with the following transformations:
+          - Converts the 'memory' field to a string in megabytes (e.g., '1024M').
+          - Converts the 'walltime' field to a string in HH:MM:SS format.
+          - Ensures the 'env' field (environment variables) is a list.
+          - Ensures the 'native' field (native scheduler directives) is a list.
+
+            A deep-copied and normalized configuration dictionary with updated fields for memory, walltime,
+            environment variables, and native scheduler directives.
+
+        Returns
+        -------
+        specs: Dict
+            A deep-copied and normalized configuration dictionary with updated fields for memory, walltime, etc.
+        """
+        specs = self._config.deepcopy()
+
+        # Convert 'memory' to MB
+        if 'memory' in self._config:
+            specs.memory = str(self.memory_in_megabytes(self._config.memory)) + 'M'
+
+        # Convert 'walltime' to HH:MM:SS format
+        if 'walltime' in self._config:
+            specs.walltime = self.walltime_in_string(self._config.walltime)
+
+        # Ensure environment variables are a list
+        if 'env' in self._config:
+            if not isinstance(self._config.env, (list, tuple)):
+                specs.env = [self._config.env]
+
+        # Ensure native scheduler directives are a list
+        if 'native' in self._config:
+            if not isinstance(self._config.native, (list, tuple)):
+                specs.native = [self._config.native]
+
+        return specs
+
+    def dump(self, filename: Optional[str] = None) -> None:
+        """
+        Dumps the batch card to a file or prints it to stdout.
+
+        If a filename is provided, writes the contents of `self.batch_card` to the specified file.
+        Otherwise, prints the batch card to standard output.
+
+        Parameters
+        ----------
+        filename : Optional[str], default=None
+            The path to the file where the batch card should be written. If None, the batch card is printed to stdout.
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        Exception
+            If an error occurs while writing to the specified file.
+        """
+        if filename is not None:
+            try:
+                with open(filename, 'w') as fh:
+                    for item in self.batch_card:
+                        fh.write(f"{item}\n")
+            except Exception as e:
+                raise f"Unknown exception in writing scheduler directives to {filename} as {e}"
```

Comment:
Exceptions do not have a `string` method, so you have to get the name via the `__name__` class attribute.
```suggestion
                raise f"Unknown exception in writing scheduler directives to {filename} as {e.__class__.__name__}"
```